### PR TITLE
OKTA-289657: Add Factors API Integration Tests

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**
### Resolves

[OKTA-289657](https://oktainc.atlassian.net/browse/OKTA-289657)

### CODE REVIEW

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/FactorsIT.groovy [hunk 1]
```diff
@@ -178,6 +188,60 @@ class FactorsIT extends ITSupport {
...
+        User user = randomUser()
+        assertThat user.listFactors(), emptyIterable()
+
+        WebUserFactor webUserFactor = client.instantiate(WebUserFactor)
+            .setFactorType(FactorType.WEBAUTHN)
+            .setProvider(FactorProvider.DUO)
+            .setProfile(client.instantiate(WebUserFactorProfile))
+
+        assertThat webUserFactor.id, nullValue()
+        assertThat webUserFactor, sameInstance(user.enrollFactor(webUserFactor)) //TODO: enable this factor in oktapreview org; until then this test will fail.
```
Comment (arvindkrishnakumar-okta) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/FactorsIT.groovy:240]: @vijetmahabaleshwar-okta May I request your help with this (assuming I don't have permissions to do the same?).

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/FactorsIT.groovy [hunk 2]
```diff
@@ -178,6 +188,60 @@ class FactorsIT extends ITSupport {
...
+        User user = randomUser()
+        assertThat user.listFactors(), emptyIterable()
+
+        U2fUserFactor u2fUserFactor = client.instantiate(U2fUserFactor)
+            .setFactorType(FactorType.U2F)
+            .setProvider(FactorProvider.GOOGLE)
+            .setProfile(client.instantiate(U2fUserFactorProfile))
+
+        assertThat u2fUserFactor.id, nullValue()
+        assertThat u2fUserFactor, sameInstance(user.enrollFactor(u2fUserFactor)) //TODO: check why Okta core responds with -  HTTP 400, Okta E0000001 (Api validation failed: factorEnrollRequest - Invalid Factor.), ErrorId oaeV_agqOzrSL-dXGwvdDLpSg
```
Comment (arvindkrishnakumar-okta) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/FactorsIT.groovy:224]: @vijetmahabaleshwar-okta Looks like the factor provider `Google` needs to be enabled in Oktapreview Org?

**Comment by arvindkrishnakumar-okta** [comment]
> Not my call, but `TokenUserFactor` isn't a great name for a TOTP factor (or any factor) as "token" is a pretty generic term in the security world.
> 
> That said, I don't remember what other token types that class supports. IIRC, we had SMS and other factors (which are all token based)

Here is the entire list of User Factor types defined in spec:

`
  x-openapi-v3-discriminator:
      mapping:
        call: '#/definitions/CallUserFactor'
        email: '#/definitions/EmailUserFactor'
        push: '#/definitions/PushUserFactor'
        question: '#/definitions/SecurityQuestionUserFactor'
        sms: '#/definitions/SmsUserFactor'
        token: '#/definitions/TokenUserFactor'
        'token:hardware': '#/definitions/HardwareUserFactor'
        'token:software:totp': '#/definitions/TotpUserFactor'
        u2f: '#/definitions/U2fUserFactor'
        web: '#/definitions/WebUserFactor'
      propertyName: factorType
`
So as per your recommendation, there shouldn't be a `TokenUserFactor` at all as it is too vague a name (given that we have more specific definitions like `HardwareUserFactor`, `TotpUserFactor` et al)? Or even better, we could have `TokenUserFactor` as the super class for the other token types? In any case, it would require a significant spec update.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/FactorsIT.groovy [hunk 3]
```diff
@@ -178,6 +189,43 @@ class FactorsIT extends ITSupport {
...
+    }
+
+    @Test
+    void testTokenUserFactorCreation() {
+        User user = randomUser()
+        assertThat user.listFactors(), emptyIterable()
+
+        TokenUserFactor tokenUserFactor = client.instantiate(TokenUserFactor)
+            .setFactorType(FactorType.TOKEN_SOFTWARE_TOTP)
+            .setProvider(FactorProvider.GOOGLE)
```
Comment (bdemers) [inline integration-tests/src/test/groovy/com/okta/sdk/tests/it/FactorsIT.groovy:221]: There is a little overlap with the `activateTotpFactor` for this one, maybe changing the name to include GoogleTotp would make it more clear (if that is the goal of this test)

**Comment by arvindkrishnakumar-okta** [comment]
> Take a look at the above comment before merging

sure, will do.

File: integration-tests/src/test/groovy/com/okta/sdk/tests/it/FactorsIT.groovy (hunk 3 above)
Comment (arvindkrishnakumar-okta) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/FactorsIT.groovy:221]: Makes sense, yeah that's the idea of this test.

Comment (arvindkrishnakumar-okta) [reply integration-tests/src/test/groovy/com/okta/sdk/tests/it/FactorsIT.groovy:221]: renamed test to `testGoogleTotpUserFactorCreation`.

**Comment by arvindkrishnakumar-okta** [comment]
> > Take a look at the above comment before merging
> 
> sure, will do.

Addressed it.

**Pull request merged**